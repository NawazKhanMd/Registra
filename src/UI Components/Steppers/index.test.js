import React from 'react';
import renderer from 'react-test-renderer';
import { Steppers } from './index';
import '@testing-library/jest-dom/extend-expect';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));

jest.mock('firebase', () => ({
  initializeApp: () => ({
    analytics: jest.fn(),
    auth: jest.fn(),
  }), firestore: jest.fn(),
}));

jest.mock('react-redux', () => {
  const ActualReactRedux = require.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return 1;
    }),
  };
});


const steps = ['Personal Info', 'Office Details', 'Confirmation']
const links = ['/', '/Office', '/Confirm']

test('Stepper Loads', () => {
  const component = renderer.create(
    <Steppers steps={steps} links={links} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('detecting the number of steps', () => {
  const wrapper = shallow((<Steppers steps={steps} links={links} />));
  expect(wrapper.find('ul').childAt(0).type()).toEqual('li');
  expect(wrapper.find('li')).toHaveLength(3);
  expect(wrapper.find('ul').childAt(0).hasClass('completed')).toEqual(true);
});
test('Simulating Click on a completed step ', () => {
  const wrapper = shallow((<Steppers steps={steps} links={links} />));
  wrapper.find('.completed').simulate('click');
  expect(wrapper.find('ul').childAt(0).hasClass('active')).toEqual(true);
});

