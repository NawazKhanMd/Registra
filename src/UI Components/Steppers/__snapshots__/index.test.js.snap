// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Loads 1`] = `
<div
  className="stepper_container"
>
  <div
    className="stepper"
  >
    <ul
      className="nav"
    >
      <li
        className="completed"
        onClick={[Function]}
      >
        <a
          className="cursor-pointer"
        >
          <span
            className="round-tab"
          >
            1
          </span>
          <span
            className="tab_title"
          >
            Personal Info
          </span>
        </a>
      </li>
      <li
        className="active"
        onClick={[Function]}
      >
        <a
          className="cursor-pointer"
        >
          <span
            className="round-tab"
          >
            2
          </span>
          <span
            className="tab_title"
          >
            Office Details
          </span>
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className="cursor-pointer"
        >
          <span
            className="round-tab"
          >
            3
          </span>
          <span
            className="tab_title"
          >
            Confirmation
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
